python/pyspark/serializers.py

```diff
@@ -296,7 +296,7 @@ def create_array(s, t):
             mask = s.isnull()
             # Ensure timestamp series are in expected form for Spark internal representation
             if t is not None and pa.types.is_timestamp(t):
-                s = _check_series_convert_timestamps_internal(s.fillna(0), self._timezone)
+                s = _check_series_convert_timestamps_internal(s, self._timezone)
```

Comment:
Removing this doesn't fail existing tests. @BryanCutler do you remember why are we doing `fillna(0)` here?

Reply:
I believe it was due to a Pandas error, most likely because we were testing with 0.19.2 at the time. Can you manually run some tests with different Pandas versions? It will be best to test with older versions, but it might be kind of hard to get 0.19.2 working with pyarrow 0.12.1 though..

Reply:
Yeah pandas 0.19.2 doesn't work with pyarrow 0.12.1. I cannot run arrow tests with pandas 0.19.2 anymore.

Since we are requiring min arrow version to be 0.12, it means pandas version 0.19.2 is not supported if the user wants to use Arrow.